robottelo/helpers.py

```diff
@@ -716,3 +716,82 @@ def idgen(val):
     test name
     """
     return '_parameter'
+
+
+class InstallerCommand:
+    """This class constructs, parses, updates and gets formatted installer commands"""
+
+    def __init__(self, *args, command='satellite-installer', allow_dupes=False, **kwargs):
+        """This allows multiple methods for InstallerClass creation
+
+        InstallerCommand('f', 'verbose', command='satellite-installer', sat_host='my_sat')
+        InstallerCommand(installer_args=['f', 'verbose'], sat_host='my_sat')
+        InstallerCommand(installer_args=['f', 'verbose'], installer_opts={'sat_host': 'my_sat'})
+
+        :param allow_dupes: Allow duplicate options, doesn't apply to future updates
+
+        """
+
+        self.command = command
+        self.args = kwargs.pop('installer_args', [])
+        self.opts = kwargs.pop('installer_opts', {})
+        self.update(*args, allow_dupes=allow_dupes, **kwargs)
+
+    def get_command(self):
+        """Construct the final command in the form of a string"""
+        command_str = self.command
+        for arg in self.args:
+            command_str += f' {"-" if len(arg) == 1 else "--"}{arg}'
+        for key, val in self.opts.items():
+            if isinstance(val, list):
+                for v in val:
+                    command_str += f' --{key.replace("_", "-")} {v}'
+            else:
+                command_str += f' --{key.replace("_", "-")} {val}'
+        return command_str
+
+    def update(self, *args, allow_dupes=False, **kwargs):
+        """Update one or more arguments and options
+        values passed as positional and keyword arguments
+        """
+        new_args = [arg for arg in args if arg not in self.args]
+        self.args.extend(new_args)
+        if not allow_dupes:
+            self.opts.update(kwargs)
+        for key, val in kwargs.items():
```

Comment:
I think this warrants some documentation. Slick little loop though, nice.

Reply:
anything you'd specifically like documented here?

Reply:
nothing specifically, just a plain language description of the intent/action of the loop so its quicker to 'read'.